### PR TITLE
added doctring to SchedulerType class

### PR DESCRIPTION
**Pull request by Arunprakash-A**
# What does this PR do?
It adds the docstring for the class `SchedulerType` . Currently, the AutoDoc only displays the `__call__ ` signature of `enum.EnumType` base class which is not helpful. One needs to look for the names and the respective mapping in the source code of trainer.py. Therefore, added the docstring that lists out all the available names and the scheduler types they are mapped to.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Documentation: @stevhliu

### CODE REVIEW

**Comment by Arunprakash-A** [comment]
@stevhliu i ran `make style` in the command line. It catched style errors in other documents, for example, "examples/research_projects/movement-pruning/Saving_PruneBERT.ipynb". I am not sure whether those should be fixed before making this pull request.

File: src/transformers/trainer_utils.py [hunk 1]
```diff
@@ -411,6 +411,21 @@ def speed_metrics(split, start_time, num_samples=None, num_steps=None, num_token
...
+       - "cosine" = get_cosine_schedule_with_warmup
+       - "cosine_with_restarts" = get_cosine_with_hard_restarts_schedule_with_warmup
+       - "polynomial" = get_polynomial_decay_schedule_with_warmup
+       - "constant" =  get_constant_schedule
+       - "constant_with_warmup" = get_constant_schedule_with_warmup
+       - "inverse_sqrt" = get_inverse_sqrt_schedule
+       - "reduce_lr_on_plateau" = get_reduce_on_plateau_schedule
+       - "cosine_with_min_lr" = get_cosine_with_min_lr_schedule_with_warmup
+       - "warmup_stable_decay" = get_wsd_schedule
+    """
```
Comment (stevhliu) [inline src/transformers/trainer_utils.py:428]: Trailing whitespace here.

```suggestion
    """
```

**Review by stevhliu — COMMENTED** [review]
Thanks!

I'm not sure why the `check_repo_consistency` test is failing since you didn't touch any of those files. cc @amyeroberts, do you know how we can fix this? I noticed the same error message in a separate PR [here](https://github.com/huggingface/transformers/pull/32893#issuecomment-2298054855).

**Review by Arunprakash-A — COMMENTED** [review]
Done

**Comment by amyeroberts** [comment]
@stevhliu @Arunprakash-A For the check_repo_consistency tests - the tests were failing because of packages on the test image. These should have been resolved now. Pushing an empty commit to trigger a new CI run should resolve. 

For `check_code_quality` - there's a formatting issue and you should run `make fixup` and push the changes to resolve

**Comment by Arunprakash-A** [comment]
> @stevhliu @Arunprakash-A For the check_repo_consistency tests - the tests were failing because of packages on the test image. These should have been resolved now. Pushing an empty commit to trigger a new CI run should resolve.
> 
> For `check_code_quality` - there's a formatting issue and you should run `make fixup` and push the changes to resolve

I ran `make fixup`  and got the following
```
Checking/fixing src/transformers/trainer_utils.py
All checks passed!
1 file reformatted

python utils/custom_init_isort.py
python utils/sort_auto_mappings.py
python utils/check_doc_toc.py --fix_and_overwrite
running deps_table_update
updating src/transformers/dependency_versions_table.py
python utils/check_copies.py
python utils/check_table.py
python utils/check_dummies.py
python utils/check_repo.py
<truncated>:
```

**Comment by amyeroberts** [comment]
@Arunprakash-A You have to commit and push the changes made when running `make fixup`

**Comment by Arunprakash-A** [comment]
> @Arunprakash-A You have to commit and push the changes made when running `make fixup`

Thanks @amyeroberts. Done!

**Review by stevhliu — APPROVED** [review]
Thanks again for improving the docstring here!

**Pull request merged**